### PR TITLE
[BUG FIX] Fix: ray version check failed due to extra output in STDOUT

### LINKED ISSUE #28222: [Core] runtime_env ray version checking fails when there's extra logging
### What happened + What you expected to happen

I have created a ray cluster and am trying to submit a job with a runtime env. The runtime env contains one package, "wandb". When creating the env, the job crashes with the following message:

```
2022-08-31 19:10:33,347 INFO packaging.py:498 -- Creating a file package for local directory './.'.
2022-08-31 19:10:33,371 INFO packaging.py:342 -- Pushing file package 'gcs://_ray_pkg_fc8fb8ff77b0621e.zip' (3.41MiB) to Ray cluster...
2022-08-31 19:10:35,482 INFO packaging.py:351 -- Successfully pushed file package 'gcs://_ray_pkg_fc8fb8ff77b0621e.zip'.
Traceback (most recent call last):
  File "/Users/zygi/llm/monorepo/models/mesh-transformer-jax/./repro.py", line 4, in <module>
    ray.init("ray://localhost:10001", runtime_env = {"working_dir": "./.", "pip":
  File "/Users/zygi/miniforge3/envs/mtj/lib/python3.10/site-packages/ray/_private/client_mode_hook.py", line 105, in wrapper
    return func(*args, **kwargs)
  File "/Users/zygi/miniforge3/envs/mtj/lib/python3.10/site-packages/ray/_private/worker.py", line 1248, in init
    ctx = builder.connect()
  File "/Users/zygi/miniforge3/envs/mtj/lib/python3.10/site-packages/ray/client_builder.py", line 178, in connect
    client_info_dict = ray.util.client_connect.connect(
  File "/Users/zygi/miniforge3/envs/mtj/lib/python3.10/site-packages/ray/util/client_connect.py", line 47, in connect
    conn = ray.connect(
  File "/Users/zygi/miniforge3/envs/mtj/lib/python3.10/site-packages/ray/util/client/__init__.py", line 252, in connect
    conn = self.get_context().connect(*args, **kw_args)
  File "/Users/zygi/miniforge3/envs/mtj/lib/python3.10/site-packages/ray/util/client/__init__.py", line 102, in connect
    self.client_worker._server_init(job_config, ray_init_kwargs)
  File "/Users/zygi/miniforge3/envs/mtj/lib/python3.10/site-packages/ray/util/client/worker.py", line 838, in _server_init
    raise ConnectionAbortedError(
ConnectionAbortedError: Initialization failure from server:
Traceback (most recent call last):
  File "/home/ray/.local/lib/python3.10/site-packages/ray/util/client/server/proxier.py", line 678, in Datapath
    if not self.proxy_manager.start_specific_server(
  File "/home/ray/.local/lib/python3.10/site-packages/ray/util/client/server/proxier.py", line 304, in start_specific_server
    serialized_runtime_env_context = self._create_runtime_env(
  File "/home/ray/.local/lib/python3.10/site-packages/ray/util/client/server/proxier.py", line 253, in _create_runtime_env
    raise RuntimeError(
RuntimeError: Failed to create runtime_env for Ray client server, it is caused by:
Traceback (most recent call last):
  File "/home/ray/.local/lib/python3.10/site-packages/ray/dashboard/modules/runtime_env/runtime_env_agent.py", line 341, in _create_runtime_env_with_retry
    runtime_env_context = await asyncio.wait_for(
  File "/usr/lib/python3.10/asyncio/tasks.py", line 445, in wait_for
    return fut.result()
  File "/home/ray/.local/lib/python3.10/site-packages/ray/dashboard/modules/runtime_env/runtime_env_agent.py", line 296, in _setup_runtime_env
    await create_for_plugin_if_needed(
  File "/home/ray/.local/lib/python3.10/site-packages/ray/_private/runtime_env/plugin.py", line 251, in create_for_plugin_if_needed
    size_bytes = await plugin.create(uri, runtime_env, context, logger=logger)
  File "/home/ray/.local/lib/python3.10/site-packages/ray/_private/runtime_env/pip.py", line 453, in create
    return await task
  File "/home/ray/.local/lib/python3.10/site-packages/ray/_private/runtime_env/pip.py", line 435, in _create_for_hash
    await PipProcessor(
  File "/home/ray/.local/lib/python3.10/site-packages/ray/_private/runtime_env/pip.py", line 331, in _run
    async with self._check_ray(python, exec_cwd, logger):
  File "/usr/lib/python3.10/contextlib.py", line 206, in __aexit__
    await anext(self.gen)
  File "/home/ray/.local/lib/python3.10/site-packages/ray/_private/runtime_env/pip.py", line 190, in _check_ray
    raise RuntimeError(
RuntimeError: Changing the ray version is not allowed: 
  current version: /usr/lib/python3/dist-packages/requests/__init__.py:89:, current path: RequestsDependencyWarning:
  expect version: 2.0.0, expect path: /home/ray/.local/lib/python3.10/site-packages/ray
Please ensure the dependencies in the runtime_env pip field do not install a different version of Ray.
```

It seems what's happening is that importing `ray` after installing `wandb` causes a logging message to get printed, and ray version checking logic doesn't handle that in https://github.com/ray-project/ray/blob/master/python/ray/_private/runtime_env/pip.py#L173

### Versions / Dependencies

python==3.10.6
ray==2.0.0
wandb==0.13.2

cluster is on gcp

### Reproduction script

import ray

ray.init("ray://localhost:10001", runtime_env = {"working_dir": "./.", "pip": 
    ["wandb"] # commenting out this line makes the code work
    })

@ray.remote
def hello():
    print("hello")

ray.get(hello.remote())

### Issue Severity

Low: It annoys or frustrates me.

### PR DESCRIPTION
**Pull request by Catch-Bull**
Signed-off-by: Catch-Bull <burglarralgrub@gmail.com>

## Why are these changes needed?

Use files to transfer information across processes

Need to discuss whether to introduce python package `asynctempfile` and `aiofiles`

## Related issue number

Closes https://github.com/ray-project/ray/issues/28222

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jjyao** [comment]
@Catch-Bull the issue you linked is actually a PR.

**Comment by architkulkarni** [comment]
> @Catch-Bull the issue you linked is actually a PR.

Thanks, I updated it

**Review by architkulkarni — COMMENTED** [review]
Thanks for the fix!  Is it feasible to add a test for this that involves some extra warnings like the ones that caused the original issue?

File: python/ray/_private/runtime_env/pip.py [hunk 1]
```diff
@@ -167,20 +168,32 @@ async def _check_ray(python: str, cwd: str, logger: logging.Logger):
...
-            if _WIN32:
-                env = os.environ.copy()
-            else:
-                env = {}
-            output = await check_output_cmd(
-                check_ray_cmd, logger=logger, cwd=cwd, env=env
-            )
-            # print after import ray may have [0m endings, so we strip them by *_
-            ray_version, ray_path, *_ = [s.strip() for s in output.split()]
+            with tempfile.TemporaryDirectory() as tmp_dir:
```
Comment (SongGuyang) [inline python/ray/_private/runtime_env/pip.py:171]: Do we need to change to use `tempfile.NamedTemporaryFile(suffix="_ray_version.txt") `?

**Review by SongGuyang — APPROVED** [review]
LGTM.

**Comment by Catch-Bull** [comment]
> Thanks for the fix! Is it feasible to add a test for this that involves some extra warnings like the ones that caused the original issue?

Hi @architkulkarni , I think maybe it doesn't need a new test case generated according to this issue. We used STDOUT to get the ray version information of another python process, and now it is changed to the file, we don't need to care about the contents of STDOUT anymore

**Review by architkulkarni — APPROVED** [review]

**Pull request merged**